### ISSUE TITLE
Screen lock while playing videos. (Ubuntu 20.04.1 + GNOME 3.36.3 + MPV 0.32.0 + Wayland)

### ISSUE THREAD

**Issue by PranavBhattarai** [mpv-player/mpv#8097]
As reported here with details: https://bugs.launchpad.net/ubuntu/+source/mpv/+bug/1895941
A similar bug was reported [here](https://github.com/mpv-player/mpv/issues/4752) and [here](https://github.com/libretro/RetroArch/issues/7472). Feel free to close this bug if it _seems_ like a duplicate.

TLDR; MPV is locking the screen while playing videos. To watch that video I have to unlock the screen which is still playing that video as if I was playing the video in the background.

```
suraj@surtoshwi ~> mpv --version
mpv 0.32.0 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
ffmpeg library versions:
   libavutil 56.31.100
   libavcodec 58.54.100
   libavformat 58.29.100
   libswscale 5.5.100
   libavfilter 7.57.100
   libswresample 3.5.100
ffmpeg version: 4.2.4-1ubuntu0.1
```

**Comment by CounterPillow** [comment]
This is GNOME's issue, they need to support the idle-inhibit wayland protocol, but they refuse to do so for political reasons.

mpv is not "locking the screen", GNOME is.

**Comment by PranavBhattarai** [comment]
For "political reasons" lol, like seriously!?
Any proof/link?

**Comment by CounterPillow** [comment]
"political reasons" as in there is no technological reason for not implementing it, they just want people to use their GNOME-specific dbus abomination instead. Feel free to look at the discussion here, in which GNOME developers argue that not supporting idle-inhibit is justified by them not supporting idle-inhibit (not a typo, just their circular logic) https://gitlab.gnome.org/GNOME/gtk/-/issues/2202

**Comment by wm4** [comment]
Yeah, GNOME being the same assbags as usual. That's why mpv does not officially support the GNOME platform. It may work somewhat, but that's really all you get, half-working garbage.

**Comment by linkmauve** [comment]
Err, you’re quite wrong, this bug report was for the GTK toolkit and [I have fixed it recently](https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/2226).

What you are looking for is this MR, which implement the idle-inhibit protocol in Mutter, and which is only blocked by the lack of people to rebase it on top of master: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/111

**Comment by wm4** [comment]
That was 2 years ago. Proof enough that GNOME folks _want_ it do die by ignoring it.
So in summary, another hint that GNOME are guilty of ruining the Linux desktop.

**Comment by PranavBhattarai** [comment]
@wm4 
I don't think it a good idea to ditch GNOME support by remembering their past behavior. 
GNOME is popular and probably the most default Desktop Environment we see in Linux distributions because of their modern look and feel.
We should look forward and at least try to solve this problem.

**Comment by CounterPillow** [comment]
> We should look forward and at least try to solve this problem.

It is literally in their hands to solve it. This is not an mpv bug.

**Comment by wm4** [comment]
And they won't fix it. If you can't do without Ubuntu/GNOME, I recommend installing Windows 10.

**Comment by Dudemanguy** [comment]
@PranavBhattarai
As a workaround, you can use `gnome-session-inhibit` when launching mpv. The mutter MR is the right way to fix it. I believe the author needs to update it for 3.38 which changed a lot of mutter/GNOME internals.

**Comment by spinninbiribin** [comment]
> Err, you’re quite wrong, this bug report was for the GTK toolkit and [I have fixed it recently](https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/2226).
> 
> What you are looking for is this MR, which implement the idle-inhibit protocol in Mutter, and which is only blocked by the lack of people to rebase it on top of master: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/111

You don't understand, you're supposed to let wm4 have another chance to rant against GNOME and propagate the narrative that they are "sabotaging things" by developing 100% free software because they don't implement an optional extension to a display server protocol he doesn't even like and which he  would have nuked already if it weren't for the other developers.

**Comment by wm4** [comment]
And you made a new github account just to say this? I've originally merged wayland support, in case you didn't know. Their slow progress in supporting fundamental desktop mechanisms (such as suspending screensaver) has left me in despair, though. Likewise, the efforts of GNOME to do their own thing and ignoring standard mechanisms (as wonderfully demonstrated by the idle inhibit thing), as well as their tendency to fuck over anyone who tries to use their stuff anyway (such as GTK compatibility and direction of development), fully confirm what I've been claiming all the time.
Thanks for giving me another opportunity to tell the truth (what you call a "rant").

**Comment by spinninbiribin** [comment]
Why doesn't mpv use real CSDs on Wayland?
Why does mpv not use XDG conventions for configuration files by default?
You disrespect standards arguably more than GNOME.

**Comment by CounterPillow** [comment]
It might be a good idea to lock conversation in this, since there is a clear path towards a resolution (that MR on GNOME's side) and now we just have (I suspect, based on their behaviour on reddit) GNOME developers making github sockpuppet accounts to rile up wm4.

> Why doesn't mpv use real CSDs on Wayland?

mpv has CSDs on Wayland.

> propagate the narrative that they are "sabotaging things" by developing 100% free software

Just because you can fork the entirety of GNOME hypothetically doesn't mean that the GNOME projects governance isn't actively sabotaging efforts for cross-compositor interoperability. You know very well that application developers aren't going to fork and maintain a massive collection of projects they're not even using.

**Comment by Dudemanguy** [comment]
>Why doesn't mpv use real CSDs on Wayland?

CSD isn't a wayland standard. And why does the lua osd not count as "real CSDs"? Anyway, agreed on locking this thread since there is a reasonable workaround for gnome users already (just use `gnome-session-inhibit`).

**Comment by PranavBhattarai** [comment]
I think two/more open source projects should go hand in hand with each other. Interoperability is the key where FOSS wins. 

Never saw this kind of arguments. 
We should respect other opinions. 
GNOME is improving. If this is wholly GNOME problem, I am sure it will solve eventually. And I expect same from this community.

But let's stop throwing rocks at each others.

**Comment by garoto** [comment]
> I think two/more open source projects should go hand in hand with each other. Interoperability is the key where FOSS wins.

That's cute, but there's no salvation for any sort of software anymore. It's a mess from top to bottom and it just keep going on and on and on, all downwards to infinity.

Have you seem how many "build" systems are involved nowadays when building ffmpeg? Yeah, it's all poop.

**Comment by fettuccine-alfredo** [comment]
Disclaimer: This is neither my opinion nor is it supposed to sound nice. This has been my experience whenever cooperation between projects has come up - both for me personally or when I've been told about it on IRC or at conferences.

Gnome doesn't do specs. If Gnome wants something to change, it goes and writes the code to change it. This is why many technologies in the Linux desktop stack were either started or are now primarily developed by people who used to work on Gnome: systemd, d-bus, pulseaudio, NetworkManager, flatpak, meson etc are all results of Gnome people wanting something and then going out and doing it. (As an aside, you can see here what Gnome cares about: If (ex-)Gnome people are part of this project, it's likely that Gnome has an interest in it - but if no Gnome people are to be found, it's likely they don't give a shit. Hello BSDs!)

Due to the above, Gnome knows lots of developers along the stack and can get them to add code they care about. Fedora is closely working together with Gnome for example, and that means that it's the perfect place to cooperate with Flatpak. It also means there's people in Mozilla, Chromium, Libreoffice, Blender, Inkscape etc who will probably be interested in adding CSD support. At least they will look at the patches.

And all of this means that Gnome has a hard time not perceiving itself as the Linux desktop. What helped this idea along quite well is the fact that Canonical just surrendered its own desktop and switched to Gnome.

TL;DR: Nobody in Gnome is gonna think Gnome or GTK owes you or mpv anything.

**Comment by mc4man** [comment]
I filed a bug on this over 2 yrs ago with mutter, lead to some patches, none ever committed, and never will be.
For those unfortunate enough to use gnome-shell, gdm3 and wayland in 20.04 mutter can be patched. Did so a month ago with cursory test, seemed ok (I don't use any of the 3 ^
Stuck in a standalone ppa, use at own risk, limited or no further support.
https://launchpad.net/~mc3man/+archive/ubuntu/mutterpatch

Or just turn off screenlock or start mpv the env=blah blah 
Not hard to find what to replace blah with

**Comment by Dudemanguy** [comment]
>people who used to work on Gnome: systemd, d-bus, pulseaudio, NetworkManager, flatpak, meson

Well out of this list, NetworkManager is the only one that's actually a Gnome project. I guess you could indirectly count d-bus since Glib is a notable implementation of it. I'm not 100% sure about the rest (and I'm too lazy to check the contributors), but I don't think any of them are particularly Gnome-heavy in contributions.

>TL;DR: Nobody in Gnome is gonna think Gnome or GTK owes you or mpv anything.

I don't think anyone is asserting this. I guess because GNOME is involved everyone has really strong opinions but as of 3.38, pretty much nearly everything mpv cares about is fixed in mutter wayland. As long as the video plays correctly (which it does), I don't really care about anything else. Implementing the idle inhibit protocol is basically the only important feature missing. But as mentioned before, there's an open MR for it. Sure, it's probably not really a priority but the basic plumbing is at least there and if it's fixed up and rebased onto 3.38, it'll probably get merged in a reasonable time frame.

Sidenote: It would be pretty nice to get presentation-time as well. Presentation time is way better for a video player than the frame callback. Well Plasma is missing that one too.

**Comment by CounterPillow** [comment]
really love this second GNOME sockpuppet account that was made today and only created for leaving that comment. Making yourself look great, guys.

**Comment by fettuccine-alfredo** [comment]
@CounterPillow
>This is GNOME's issue, they need to support the idle-inhibit wayland protocol, but they refuse to do so for political reasons.

Your idea of collaboration is small projects doing nothing but demand things and GNOME working for them.

>"political reasons" as in there is no technological reason for not implementing it, they just want people to use their GNOME-specific dbus abomination instead. Feel free to look at the discussion here, in which GNOME developers argue that not supporting idle-inhibit is justified by them not supporting idle-inhibit (not a typo, just their circular logic)

You're doing it again. You're arguing that GNOME do work for others out of some sense of entitlement.

And you invent a bunch of lies for why that should be the case. There is not a single true statement in your whole post.
Which is the best proof for how GNOME is doing the right thing: You cannot find a single reason for why GNOME is wrong. You have to make shit up and fantasize about RFCs.

@wm4
>That was 2 years ago. Proof enough that GNOME folks want it do die by ignoring it.
So in summary, another hint that GNOME are guilty of ruining the Linux desktop.

Every single major distro ships the Gnome desktop as the default (or only supported) desktop.

Gnome people cooperate all the time - with Wayland, with X, with systemd, with whatever.
They don't cooperate with people who don't matter.

>Yeah, GNOME being the same assbags as usual. That's why mpv does not officially support the GNOME platform. It may work somewhat, but that's really all you get, half-working garbage.

Stop doing shit that doesn't matter to Gnome (like whining on reddit) and do something that people actually care about.

But until then, this entitlement of thinking other people should do work to make the things happen that you think should happen is the bullshit that needs to end. You don't do shit, so you have nothing to say.

It leads to people either showing up or putting up. Which is way better than the constant whining because Gnome is too nice to tell people like you that you're just a whiny little kid whose opinion doesn't matter.

@Dudemanguy
>Sidenote: It would be pretty nice to get presentation-time as well. Presentation time is way better for a video player than the frame callback. Well Plasma is missing that one too.

It's weird that nothing ever gets done if Gnome isn't involved, isn't it?

**Comment by Dudemanguy** [comment]
>It's weird that nothing ever gets done if Gnome isn't involved, isn't it?

Sway/wlroots and weston have already implemented it.

**Comment by CounterPillow** [comment]
If we don't matter why are you on our issue tracker?

**Comment by Hrxn** [comment]
You know who's not a sockpuppet? Guaranteed. Check my account creation date.

So, allow me this opportunity to step in for a second, to bring a perspective to this exchange that has been sorely missing so far.

I want to say this, in defense of the GNOME project.




No, wait.



I don't. Fuck these guys.

> It leads to people either showing up or putting up. Which is way better than the constant whining because Gnome is too nice to tell people like you that you're just a whiny little kid whose opinion doesn't matter.

You know, you can pull this crap with someone like me, who is just a simple user. I don't matter here in the grand scheme of things. But to come here, and shit on the contributors of the mpv project, you know, like wm4 and so on, guys who *actually put in time and do work here*, really takes some gall.
Bad optics, you GNOMEs, bad optics.

**Comment by Akemi** [comment]
i am stopping this bullshit now.

**Comment by wm4** [comment]
> systemd, d-bus, pulseaudio, NetworkManager, flatpak, meson etc are all results of Gnome people wanting something and then going out and doing it.

It seems you are confusing RedHat with Gnome. Other than that, it's bullshit of course. I also used Gnome at some point, is mpv a result of a Gnome person wanting something and then going out and doing it? Excellent trolling, though.

> Every single major distro ships the Gnome desktop as the default (or only supported) desktop.

And that is why the Linux desktop has this pathetic reputation it does have.
